pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -443,6 +445,10 @@ public CompletableFuture<Void> messageAcked(CommandAck ack) {
             properties = ack.getPropertiesList().stream()
                 .collect(Collectors.toMap(KeyLongValue::getKey, KeyLongValue::getValue));
         }
+        if (readCompacted) {
+            properties = new TreeMap<>(properties);
+            properties.put(READ_COMPACTED_CURSOR_PROPERTIES, 1L);
+        }
```

Comment:
It's meaningless to use a TreeMap here.

```suggestion
        if (readCompacted) {
            properties = Collections.singletonMap(ManagedCursorImpl.READ_COMPACTED_CURSOR_PROPERTIES, 1L);
        }
```

Reply:
Using `HashMap` instead.